docs/commerce/payment/payment_method_filtering.md

```diff
@@ -0,0 +1,56 @@
+---
+description: Implement payment method filtering.
+edition: commerce
+---
+
+# Implement payment method filtering
+
+You can use payment method filtering to decide, whether selected payment method can be used and displayed for specific product types in checkout process.
+To allow this filtering, you need to create custom payment method type and register new voter.
+
+## Create custom payment method type
+
+There are different ways you can extend your Payment module implementation. 
+One of them is to [create a custom payment method type](extend_payment.md). 
+
+The following example shows, how to create `New Payment Method Type`.
+
+### Add new custom payment method type
+
+First, in `src/bundle/Resources/config/services` directory register new payment method type as a service:
+
+``` yaml
+[[= include_file('code_samples/front/shop/payment/src/bundle/Resources/config/services/payment_method.yaml', 0, 9) =]]
+```
+
+In the `arguments` list provide a name of the payment method type, the way you want it to appear on the list of available payment method types, in the following example: `New Payment Method Type`.
+
+At this point a custom payment method type should be visible in the user interface.
+
+### Create voter for new payment method type
+
+Now, in `src/lib/PaymentMethod/Voter` directory, create a `NewPaymentMethodTypeVoter.php` file with the voter definition for your new payment method type:
+
+``` php
+[[= include_file('code_samples/front/shop/payment/src/src/lib/PaymentMethod/Voter/NewPaymentMethodTypeVoter.php') =]]
+```
+
+Created voter decides, if selected payment method type can be used and displayed for specific product types in checkout process.
+In the following example, when cart contains only virtual products all offline payment methods should not be visible on payment method list.
```

Comment:
IMO this should be changed, because it is not clear.
There is a no custom logic in provided example, which makes "New Payment Method Type" always visible in checkout.